src/tinuous/github.py

```diff
@@ -132,16 +134,41 @@ def get_event_id(self, run: WorkflowRun, event_type: EventType) -> str:
                     # have to fall back to performing an issue search to fill
                     # those in.  This should hopefully be used sparingly, as
                     # there's a 30 searches per hour rate limit.
-                    try:
-                        pr = str(
-                            self.client.search_issues(
-                                f"repo:{run.repository.full_name} is:pr {run.head_sha}",
-                                sort="created",
-                                order="asc",
-                            )[0].number
-                        )
-                    except IndexError:
-                        pr = "UNK"
+                    while True:
+                        try:
+                            pr = str(
+                                self.client.search_issues(
+                                    f"repo:{run.repository.full_name} is:pr"
+                                    f" {run.head_sha}",
+                                    sort="created",
+                                    order="asc",
+                                )[0].number
+                            )
+                            break
+                        except IndexError:
+                            pr = "UNK"
+                            break
+                        except RateLimitExceededException:
```

Comment:
is this the only point where we talk to github API directly? if not -- I think it is worth adding a helper or moving this deeper within client so we could use the same logic for any other API call -- we might hit those limits for other queries as well

Reply:
It's the only place we use the search API, which has a 30-request limit (as opposed to normal requests, which have a limit of 5000).

Reply:
I get that.  But even with 5000 limit for others, we might hit them, thus I thought that a more generic solution would be better since would immediately resolve any other similar issue.  But ok, let's proceed with only the one for `/search` and RF whenever someone hits any other limit.